Dockerfile

```diff
@@ -0,0 +1,28 @@
+FROM python:3.10.4-buster as base
+
+RUN pip install poetry
+
+WORKDIR /app
+
+COPY pyproject.toml poetry.toml poetry.lock* /app/
+
+RUN poetry install
+
+EXPOSE 5000
+
+COPY . /app
+
+FROM base as production
+
+ENV FLASK_ENV=production
+
+CMD ["poetry", "run", "flask", "run", "--host=0.0.0.0"]
```

Comment:
It might be worth adding newlines between build stages so that they are easier to spot in the code.